tools/cp-caps/device_manager.py

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env python3
+#
+#  Copyright (c) 2025, The OpenThread Authors.
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import logging
+import os
+import sys
+import threading
+import queue
+
+from dataclasses import dataclass
+from typing import Optional
+
+import otci
+from otci import OTCI
+from otci.errors import ExpectLineTimeoutError, InvalidArgumentsError
+
+CP_CAPABILITY_VERSION = "0.1.1-dev"
+
+logging.basicConfig(level=logging.WARNING)
+
+
+@dataclass
+class Frame:
+    """Represents a Thread radio frame.
+
+    Attributes:
+      name: The description of the frame.
+      tx_frame: The psdu of the frame that to be sent.
+      dst_address: The destination MAC address of the tx_frame. It is used by the receiver to filter
+          out the tx_frame.
+      is_security_processed: The value of the otRadioFrame.mInfo.mTxInfo.mIsSecurityProcessed field.
+          If it is set to False, the active_dataset and src_ext_address are should also be set for the
+          radio driver to encrypt the tx_frame.
+      expect_rx_frame: The frame expected to be received. The frame expected to be received should be
```

Comment:
```suggestion
      expect_rx_frame: The frame expected to be received. The expected frame should be
```

Separate comment: Should "should" be "will" here?  Is it always the same or are there scenarios where it's not?

Reply:
It is always the same if the `expect_rx_frame ` is not None. It is `should` here.  I have updated the comment.